seed/building_sync/building_sync.py

```diff
@@ -6,229 +6,146 @@
+
+    def export(self, property_state, custom_mapping=None):
         """Export BuildingSync file from an existing BuildingSync file (from import), property_state and
-        a process struct.
+        a custom mapping.
 
         :param property_state: object, PropertyState to merge into BuildingSync
-        :param process_struct: dict, mapping from PropertyState to BuildingSync
+        :param custom_mapping: dict, user-defined mapping (used with higher priority over the default mapping)
         :return: string, as XML
         """
-
-        # if property state is not defined, then just return the BuildingSync unparsed
         if not property_state:
-            return xmltodict.unparse(self.raw_data, pretty=True).replace('\t', '  ')
-
-        # parse the property_state and merge it with the raw data
-        new_dict = copy.deepcopy(self.raw_data)
-        if new_dict == {}:
-            _log.debug("BuildingSync raw data is empty, adding in header information")
-            # new_dict[]
-            new_dict = OrderedDict(
-                [
-                    (
-                        'auc:BuildingSync', OrderedDict(
-                            [
-                                ('@xsi:schemaLocation',
-                                 'http://buildingsync.net/schemas/bedes-auc/2019 https://github.com/BuildingSync/schema/releases/download/v1.0.0/BuildingSync.xsd'),
-                                ('@xmlns', OrderedDict(
-                                    [
-                                        ('auc', 'http://buildingsync.net/schemas/bedes-auc/2019'),
-                                        ('xsi', 'http://www.w3.org/2001/XMLSchema-instance')
-                                    ]
-                                ))
-                            ]
-                        )
-                    )
-                ]
-            )
-        else:
-            # check that the appropriate headers are set or XML won't render correctly in the browser
-            if '@xsi:schemaLocation' not in new_dict['auc:BuildingSync'] or '@xmlns' not in new_dict['auc:BuildingSync']:
-                new_dict['auc:BuildingSync']['@xsi:schemaLocation'] = 'http://buildingsync.net/schemas/bedes-auc/2019 https://github.com/BuildingSync/schema/releases/download/v1.0.0/BuildingSync.xsd'
-                new_dict['auc:BuildingSync']['@xmlns'] = OrderedDict
-                (
-                    [
-                        ('auc', 'http://buildingsync.net/schemas/bedes-auc/2019'),
-                        ('xsi', 'http://www.w3.org/2001/XMLSchema-instance')
-                    ]
-                )
-
-        for field, v in process_struct['return'].items():
+            return etree.tostring(self.element_tree, pretty_print=True).decode()
+
+        if not self.element_tree:
+            self.init_tree(version=BuildingSync.BUILDINGSYNC_V2_0)
+
+        merged_mappings = merge_mappings(self.VERSION_MAPPINGS_DICT[self.version], custom_mapping)
```

Comment:
That's a good question, I can think of two cases. For the export tests I don't pass a custom mapping because it's simpler. Also, if a user has never imported a buildingsync file, but wants to export one of their properties as buildingsync, they will not have a mapping saved (this happens only when importing a bsync file), so their custom mapping would be `None` as well.